src/main/scala/ai/privado/languageEngine/php/tagger/collection/ConfigCollectionTagger.scala

```diff
@@ -0,0 +1,188 @@
+package ai.privado.languageEngine.php.tagger.collection
+
+import ai.privado.cache.RuleCache
+import ai.privado.languageEngine.java.tagger.collection.CollectionUtility
+import ai.privado.model.{CatLevelOne, Constants, FilterProperty, Language, NodeType, RuleInfo}
+import ai.privado.tagger.PrivadoParallelCpgPass
+import better.files.File.VisitOptions
+import io.joern.x2cpg.SourceFiles
+import io.shiftleft.codepropertygraph.generated.Cpg
+import org.slf4j.LoggerFactory
+import io.shiftleft.semanticcpg.language.*
+
+import scala.collection.mutable
+import io.circe.*
+import io.circe.generic.auto.*
+import io.circe.yaml
+import io.shiftleft.codepropertygraph.generated.nodes.Method
+
+case class Route(path: String, defaults: Option[Defaults], controller: Option[String])
+case class Defaults(_controller: String)
+
+case class ControllerAlias(clazz: String)
+
+class ConfigCollectionTagger(cpg: Cpg, ruleCache: RuleCache, projectRoot: String)
+    extends PrivadoParallelCpgPass[Route](cpg) {
+
+  private val configExtensions: Set[String]                 = Set(".yaml", ".yml")
+  protected val methodUrlMap: mutable.HashMap[Long, String] = mutable.HashMap[Long, String]()
+  protected val classUrlMap: mutable.HashMap[Long, String]  = mutable.HashMap[Long, String]()
+  private val logger                                        = LoggerFactory.getLogger(this.getClass)
+
+  override def generateParts(): Array[Route] = {
+    SourceFiles
+      .determine(projectRoot, configExtensions, ignoredFilesRegex = Some(".*[.]privado.*".r))(VisitOptions.default)
+      .filter(_.contains(Constants.routes))
+      .flatMap(getRoutesFromConfig)
+      .toArray
+  }
+
+  override def runOnPart(builder: DiffGraphBuilder, route: Route): Unit = {
+    val (controllerClassName, controllerMethodName) = getClassAndMethodNameFromControllerFullName(
+      route.controller
+        .getOrElse(findControllerFullNameFromAlias(route.defaults.getOrElse(Defaults(""))._controller))
+    )
+
+    methodUrlMap.addOne(
+      (
+        cpg.method.nameExact(controllerMethodName).headOption match {
+          case Some(method) => method.id()
+          case None =>
+            cpg.typeDecl.nameExact(controllerClassName).astChildren.isMethod.nameExact("__invoke").headOption match {
+              case Some(invokeMethod) => invokeMethod.id()
+              case None               => -1
+            }
+        },
+        route.path
+      )
+    )
+
+    // The RuleInfo is intended to be empty
+    tagSources(
+      builder,
+      RuleInfo(
```

Comment:
Why are we passing the empty ruleInfo here, please elaborate.

Also if we need to pass empty ruleInfo, better to create a helper function on top of `object RuleInfo` which facilitates this